src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -377,8 +377,6 @@ private void WriteUnnamedBuildCommand(CodeMethod codeElement, LanguageWriter wri
 
         protected virtual void WriteCommandHandlerBody(CodeMethod codeElement, RequestParams requestParams, bool isVoid, string returnType, LanguageWriter writer)
         {
-            if (codeElement.HttpMethod == null) throw new InvalidOperationException("http method cannot be null");
```

Comment:
I'm curious to why you're removing this defensive programming statement since you need the property value at the next line?

Reply:
Because it'll never be null. The check is done in the calling site. See L43 to L60

Reply:
thanks for the context!